extensions/amp-analytics/0.1/events.js

```diff
@@ -414,6 +421,29 @@ export class VisibilityTracker extends EventTracker {
   }
 
   /**
+   * @return {!Promise}
+   * @visibleForTesting
+   */
+  createReportReadyPromise_() {
+    const viewer = this.root.getViewer();
+    let resolver = null;
```

Comment:
Remove `let resolver` from here and move the `new Promise` and recombine with the listener below. Ideally, the less we leak resolve/reject methods the better. E.g.

```
if (!isVisible()) {
  return Promise.resolve();
}
return new Promise(resolve => {
  viewer.onVisibilityChanged(() => {
    if (!isVisible()) {
      resolve();
    }
  });
});
```